pkg/nas/accesspoint_controller.go

```diff
@@ -0,0 +1,268 @@
+	if basePath = parameters["path"]; basePath == "" {
+		basePath = "/"
+	} else {
+		basePath = filepath.Clean(basePath)
+	}
+
+	// create accesspoint
+	result, err := c.createAccesspoint(ctx, req.Name, basePath, cnfs, parameters)
+	if err != nil {
+		return nil, err
+	}
+
+	volumeContext := map[string]string{}
+	volumeContext["containerNetworkFileSystem"] = cnfs.Name
+	volumeContext["path"] = "/"
+	volumeContext["accesspoint"] = tea.StringValue(result.Body.AccessPoint.AccessPointDomain)
+	volumeContext["accesspointId"] = tea.StringValue(result.Body.AccessPoint.AccessPointId)
+	if enableRam := parameters["accesspointEnableRam"]; enableRam != "" {
+		volumeContext["accesspointEnableRam"] = enableRam
+	}
+	resp := &csi.CreateVolumeResponse{
+		Volume: &csi.Volume{
+			VolumeId:      req.Name,
+			VolumeContext: volumeContext,
+		},
+	}
+
+	// set dir quota
+	if parameters["volumeCapacity"] == "true" || parameters["allowVolumeExpansion"] == "true" {
+		capacity := req.GetCapacityRange().GetRequiredBytes()
+		quota := (capacity + GiB - 1) >> 30
+		resp.Volume.CapacityBytes = quota << 30
+		resp.Volume.VolumeContext["volumeCapacity"] = "true"
+		if err := c.nasClient.SetDirQuota(&sdk.SetDirQuotaRequest{
+			FileSystemId: &cnfs.Status.FsAttributes.FilesystemID,
+			Path:         tea.String(path.Join(basePath, req.Name)),
+			SizeLimit:    &quota,
+			QuotaType:    tea.String("Enforcement"),
+			UserType:     tea.String("AllUsers"),
+		}); err != nil {
+			return nil, status.Errorf(codes.Internal, "nas:SetDirQuota failed: %v", err)
+		}
+	}
+	return resp, nil
+}
+
+func (c *accesspointController) createAccesspoint(ctx context.Context, name, basePath string, cnfs *cnfsv1beta1.ContainerNetworkFileSystem, parameters map[string]string) (*sdk.CreateAccessPointResponse, error) {
+	filesystemId := cnfs.Status.FsAttributes.FilesystemID
+	if filesystemId == "" {
+		return nil, status.Error(codes.InvalidArgument, "missing filesystemId in CNFS status")
+	}
+	// Only standard filesystems support AccessPoint.
+	filesystemType := cnfs.Status.FsAttributes.FilesystemType
+	if filesystemType != cloud.FilesystemTypeStandard {
+		return nil, status.Error(codes.InvalidArgument, "only filesystems of standard type support accesspoint")
+	}
+	vpcId := cnfs.Status.FsAttributes.VpcID
+	if vpcId == "" {
+		return nil, status.Error(codes.InvalidArgument, "missing vpc id in CNFS status")
+	}
+	vswId := cnfs.Status.FsAttributes.VSwitchID
+	if vswId == "" {
+		return nil, status.Error(codes.InvalidArgument, "missing vSwitch id in CNFS status")
+	}
+
+	req := &sdk.CreateAccessPointRequest{
+		AccessGroup:     tea.String(cloud.DefaultAccessGroup),
+		FileSystemId:    &filesystemId,
+		VpcId:           &vpcId,
+		VswId:           &vswId,
+		AccessPointName: &name,
+		RootDirectory:   tea.String(path.Join(basePath, name)),
+		OwnerUserId:     tea.Int32(0),
+		OwnerGroupId:    tea.Int32(0),
+	}
+
+	var invalidParameters []string
+	parseParameter := func(key string, target any) {
```

Comment:
elegant 👍

Reply:
I would prefer to just switch on key. So that we can report errors on unknown keys. And we can get a compilation error when unknown value type is encountered (maybe due to future SDK update).